subprojects/test-runner/device-provider/impl/src/main/kotlin/com/avito/android/runner/devices/internal/kubernetes/KubernetesApi.kt

```diff
@@ -94,63 +94,48 @@ internal interface KubernetesApi {
             }
         }
 
-        override suspend fun getPods(deploymentName: String): Result<List<Pod>> = Result.tryCatch {
+        private fun failFastOnBadImage(pods: List<KubePod>) {
+            pods.forEach { pod ->
+                val phase = pod.container.phase
+                if (phase is KubeContainer.ContainerPhase.Waiting) {
+                    if (phase.hasProblemsGettingImage()) {
+                        error("Can't create pods for deployment, check image reference: ${phase.message}")
```

Comment:
Could we return the answer to the caller and make him do the decision?

Reply:
yes, that's better

Reply:
It's already returned to a client as a `Result.Failure<>` will make this aspect more readable in next PR